File: frontend/src/api/transcripts.ts
import { get, post } from "./client";

export interface TranscriptSubmitRequest {
  title: string;
  team_name?: string;
  content: string;
}

export interface TranscriptResponse {
  id: string;
  title: string;
  team_name: string | null;
  status: string;
  content_hash: string;
  created_at: string;
}

export interface PaginatedTranscriptResponse {
  items: TranscriptResponse[];
  total: number;
  offset: number;
  limit: number;
}

export interface TranscriptSummaryResponse {
  id: string;
  transcript_id: string;
  executive_summary: string;
  action_items: string[];
  key_decisions: string[];
  dependencies: string[];
  blockers: string[];
  sentiment: string | null;
  attestation_available: boolean;
  created_at: string;
  policy_report: any | null;
  policy_constraints: any[] | null;
  egress_log: any[] | null;
  verification: any | null;
}

export interface AggregationResponse {
  cross_team_summary: string;
  shared_dependencies: string[];
  shared_blockers: string[];
  recommendations: string[];
  transcript_count: number;
  attestation_available: boolean;
  verification: any | null;
}

export const submitTranscript = (data: TranscriptSubmitRequest) => post<TranscriptResponse>("/transcripts/", data);
export const listTranscripts = (offset = 0, limit = 25) => get<PaginatedTranscriptResponse>(`/transcripts/?offset=${offset}&limit=${limit}`);
export const getTranscript = (id: string) => get<TranscriptResponse>(`/transcripts/${id}`);
export const getSummary = (id: string) => get<TranscriptSummaryResponse>(`/transcripts/${id}/summary`);
export const aggregate = (ids: string[]) => post<AggregationResponse>("/transcripts/aggregate", { transcript_ids: ids });

File: ndai/api/schemas/transcripts.py
"""Pydantic schemas for Props (meeting transcript processing)."""

from datetime import datetime

from pydantic import BaseModel, Field


class TranscriptSubmitRequest(BaseModel):
    title: str = Field(..., min_length=1, max_length=500)
    team_name: str | None = None
    content: str = Field(..., min_length=10)


class TranscriptResponse(BaseModel):
    id: str
    title: str
    team_name: str | None
    status: str
    content_hash: str
    created_at: datetime


class TranscriptSummaryResponse(BaseModel):
    id: str
    transcript_id: str
    executive_summary: str
    action_items: list[str]
    key_decisions: list[str]
    dependencies: list[str]
    blockers: list[str]
    sentiment: str | None
    attestation_available: bool = True
    created_at: datetime
    policy_report: dict | None = None
    policy_constraints: list[dict] | None = None
    egress_log: list[dict] | None = None
    verification: dict | None = None


class AggregationRequest(BaseModel):
    transcript_ids: list[str] = Field(..., min_length=2)


class AggregationResponse(BaseModel):
    cross_team_summary: str
    shared_dependencies: list[str]
    shared_blockers: list[str]
    recommendations: list[str]
    transcript_count: int
    attestation_available: bool = True
    verification: dict | None = None
